### PR TITLE
Check needinfo for errors before submitting order

### PR DESCRIPTION
**Pull request by OzzieOrca**
I added basic error messages but they aren't styled very well and I'm not sure the best place to display them. I've messaged Sudarma to get his thoughts for showing generic error messages. I'm not sure who else I should talk to about UX.

### CODE REVIEW

**Comment by adammeyer** [comment]
@OzzieOrca David is the right guy.

File: src/common/services/api/order.service.js [hunk 1]
```diff
@@ -179,6 +181,21 @@ class Order{
...
+      path: ['carts', this.cortexApiService.scope, 'default'],
+      zoom: {
+        needInfo: 'order:needinfo'
+      }
+    })
+      .map((data) => {
+        return map(data.needInfo, 'name');
+      })
+      .do((errors) => {
+        this.$log.error('The user was presented with these `needinfo` errors. They should have been caught earlier in the checkout process.', errors);
```
Comment (adammeyer) [inline src/common/services/api/order.service.js]: Is it standard practice to log errors in production?

Comment (OzzieOrca) [inline src/common/services/api/order.service.js]: My thought was to pipe `$log` to rollbar eventually but I've never actually done that. Do you guys have any thoughts?

Comment (adammeyer) [inline src/common/services/api/order.service.js]: Good idea.  I like it.

**Comment by adammeyer** [comment]
👍

**Comment by OzzieOrca** [comment]
I've updated this and moved the error messages to the top.

**Comment by Omicron7** [comment]
:thumbsup:

**Pull request merged**